src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
+  return sum / children.length;
```

Comment:
I think the difference A mother's age at child birth is not taken into account here. I think the code should look like this
```suggestion
  const filteredMotherChildPairs = onlyWithSon
    ? people
        .filter((person) => person.sex === 'f')
        .filter((mother) =>
          people.some(
            (child) =>
              child.mother === mother.name &&
              child.sex === 'm'
          )
        )
    : people
        .filter((person) => person.sex === 'f')
        .filter((mother) =>
          people.some(
            (child) => child.mother === mother.name
          )
        );

  const ageDiffs = filteredMotherChildPairs.map((mother) => {
    const child = people.find(
      (child) =>
        child.mother === mother.name &&
        (onlyWithSon ? child.sex === 'm' : true)
    );

    return child ? child.born - mother.born : 0;
  });

  const totalAgeDiff = ageDiffs.reduce(
    (accumulator, ageDiff) => accumulator + ageDiff,
    0
  );

  return ageDiffs.length === 0 ? 0 : totalAgeDiff / ageDiffs.length;
```